File: 162.py
class Solution:
    def findPeakElement(self, nums: list[int]) -> int:
        l = 0
        r = len(nums) - 1
        while l <= r:
            m = (r - l) // 2 + l
            if m > 0 and nums[m] < nums[m - 1]:
                r = m - 1
            elif m < len(nums) - 1 and nums[m] < nums[m + 1]:
                l = m + 1
            else:
                return m
        return l

File: 162.cpp
#include <algorithm>
#include <deque>
#include <iostream>
#include <map>
#include <set>
#include <unordered_map>
#include <unordered_set>
#include <vector>

using namespace std;
class Solution {
public:
  int findPeakElement(vector<int> &nums) {
    int n = nums.size();
    if (n == 1) {
      return 0;
    }
    if (nums[0] > nums[1]) {
      return 0;
    }
    for (int i = 1, j = 0; i < n; ++i, ++j) {
      if (nums[i] < nums[j]) {
        return j;
      }
    }
    return n - 1;
  }
};
int main() {}